src/mds/MDSRank.cc

```diff
@@ -3847,3 +3854,27 @@ void MDSRank::send_task_status() {
 
   schedule_update_timer_task();
 }
+
+void MDSRank::schedule_inmemory_logger() {
+  timer.add_event_after(g_conf().get_val<double>("mds_extraordinary_events_dump_interval"),
+                        new LambdaContext([this]() {
+                          inmemory_logger();
+                        }));
+}
+
+void MDSRank::inmemory_logger() {
+  if (g_conf().get_val<double>("mds_extraordinary_events_dump_interval") &&
+      (client_eviction_dump || beacon.missed_beacon_ack_dump ||
+       beacon.missed_internal_heartbeat_dump || state < MDSMap::STATE_ACTIVE)) {
```

Comment:
Doesn't accessing `state` require acquiring `Beacon::mutex`?

Reply:
I've checked using `state` (`mds->get_state()`) in other places like MDSRankDispatcher, MDCache, MDSTableClient.cc etc. It doesn't seem acquiring `Beacon::mutex`.

Reply:
The `state` is already under `mds_lock`.